backend/src/main/scala/cromwell/backend/standard/StandardAsyncExecutionActor.scala

```diff
@@ -1096,7 +1096,12 @@ trait StandardAsyncExecutionActor
             case Success(returnCodeAsInt) if isAbort(returnCodeAsInt) =>
               Future.successful(AbortedExecutionHandle)
             case Success(returnCodeAsInt) if !continueOnReturnCode.continueFor(returnCodeAsInt) =>
-              val executionHandle = Future.successful(FailedNonRetryableExecutionHandle(WrongReturnCode(jobDescriptor.key.tag, returnCodeAsInt, stderrAsOption), Option(returnCodeAsInt)))
+              // Read stderr content only if there is need to get detailed info about failure
+              // Should not read large files completely
+              val errorMessageAsFutureString = asyncIo.contentAsStringAsync(stderr, maxBytes = Option(1024), failOnOverflow = false)
```

Comment:
Maybe nitpicking but I believe it's better to use `Some.apply` in cases when we sure it can't be a `None`.

Reply:
I see they use `Option.apply` everywhere so just ignore this comment.